src/test/java/com/vinyl/AddUserTest.java

```diff
@@ -0,0 +1,97 @@
+package com.vinyl;
+
+import com.vinyl.model.User;
+import com.vinyl.modelDTO.UserDTO;
+import com.vinyl.repository.UserRepository;
+import com.vinyl.service.PasswordEncoder;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import static org.mockito.ArgumentMatchers.any;
+
+public class AddUserTest extends BaseIntegration {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private BCryptPasswordEncoder originalPasswordEncoder;
+
+    private BCryptPasswordEncoder mockedPasswordEncoder;
+
+    UserDTO userDTO = new UserDTO();
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        setUserDTO(userDTO);
```

Comment:
do the new UserDTO() in the method and return the dto, do the assignment here, also rename method to createUserDTO()

Reply:
done